Added /roll to (dev)capabilities

arichards611 opened the pull request:
Added empty string handling for /roll 
If roll w/o arguements, you get help blurb
Updated /halp to include /roll
Added test for /halp since none
Added more /roll tests
Fixed oops in roll-int where int was left negative

arichards611 committed:
Added /roll to (dev)capabilities
Added empty string handling for /roll, if roll w/o arguements, you get help blurb
updated /halp to include /roll
added tests for /halp since none, added more /roll tests

arichards611 committed:
Added some comments for clarity
if /roll is 0 is no longer "if <=" and is now "if =="

arichards611 committed:
Changed /roll handling for 0 and 1 because I'm being picky with logic.

gelstudios commented on dice.py:
```diff
@@ -3,15 +3,18 @@
+            if stuff.isdigit(): # If the string is all digits, will return true
+                size = int(stuff) # Because stuff is a string, we need a integer
+                if size == 0 or 1: # If the integer is 0 or 1
+                    result = "Cannot roll "+str(size)+", bro. Come on, think of a "+str(size)+" sided die."
```
this is great! I should point out that string interpolation is preferable to doing "string" + "thing".

ex: `"this is a {0} bot, bro".format("squanchy")` returns `"this is a squanchy bot, bro"`

The number inside the `{ }` indicates what position argument to replace with `.format()` method. You can also use a position more than once. 

ex: `"ants {0} in my {0}  {1} eyes".format( "not" , "fleeb" )`  returns
 `"ants not in my not fleeb eyes"`.

FIX IT :-1:

gelstudios commented on dev_capabilities.json:
```diff
@@ -37,6 +37,10 @@
             "event" : "room_message" } ],
+            {
+            "pattern" : "^/roll .+",
+            "url" : "http://imgur-hipchat.herokuapp.com",
```
the `dev_capabilities.json` file requires that all hostnames point to the staging env: `imgur-hipchat-dev`, otherwise the web hook will be sent to the wrong place(!)

gelstudios commented on capabilities.json:
```diff
@@ -37,6 +37,10 @@
             "url" : "http://imgur-hipchat.herokuapp.com",
             "event" : "room_message" } ],
+            {
+            "pattern" : "^/roll .+",
```
Do you want hipchat to send dankBot messages that start with `/roll` or only messages that contain something afterwards like `/roll stuff here` ?

Because if you look at the pattern for `/halp`, you'll see it looks a little different.

arichards611 committed:
Fixed capabilities
Dev now points to dev (I made a copy paste mistake)
Changed string for 0 and 1 for interpolation

arichards611 committed:
Forgot to remove text requirement for /roll from dev
